lib/faraday_middleware/gzip.rb

```diff
@@ -48,7 +48,17 @@ def uncompress_gzip(body)
     end
 
     def inflate(body)
+      # Inflate as a DEFLATE (RFC 1950+RFC 1951) stream
       Zlib::Inflate.inflate(body)
+    rescue Zlib::DataError
```

Comment:
I'll be perfectly fine to keep this part as pure as it is now if it is acceptable to drop `deflate` from the default Accept-Encoding value.  My focus is simply to prevent the error that wouldn't occur if it wasn't for the Gzip middleware enabled.